core/cache/watcher.go

```diff
@@ -131,3 +131,98 @@ func (w *ConfigWatcher) configChanged(topic string, value interface{}) {
 	}
 	w.notify()
 }
+
+// StringsWatcher will return what has changed.
+type StringsWatcher interface {
+	Watcher
+	Changes() <-chan []string
+}
+
+type stringsWatcherBase struct {
+	tomb    tomb.Tomb
+	changes chan []string
+	// We can't send down a closed channel, so protect the sending
+	// with a mutex and bool. Since you can't really even ask a channel
+	// if it is closed.
+	closed bool
+	mu     sync.Mutex
+}
+
+func newStringsWatcherBase(values ...string) *stringsWatcherBase {
+	// We use a single entry buffered channel for the changes.
+	// This allows the config changed handler to send a value when there
+	// is a change, but if that value hasn't been consumed before the
+	// next change, the second change is discarded.
+	ch := make(chan []string, 1)
```

Comment:
Should this be longer?  Right now, the code for sending changes, doesn't take into consideration that the prior change hasn't been grabbed yet.  However the end user of WatchMachines will require all changes.

Reply:
So sending more values on the same buffered channel could potentially cause a deadlock, unless it's constantly grabbing the changes. If we know that the reads are congruent to the writes, this doesn't need buffering, it depends if you want to block the send.

Reply:
I don't think we can implement this in the same way as the notify watcher (with a one-buffered channel). That works OK for notify, because we know that when the watch initiator peels the event off the channel and queries, it will get the data that caused any of the events that we discarded.

Here, discarding events means missing changes.

Perhaps we need to to implement a collect method to compress events while channel writes are blocked waiting for the other side to read, similar to the state-based strings watcher.